src/decays/H_SM_decays/decay_AH_to_dqdq.cpp

```diff
@@ -0,0 +1,130 @@
+
+// specialization for the AH -> Fd Fd case
+
+template<>
+double CLASSNAME::get_partial_width<AH, bar<dq>::type, dq>(
+   const context_base& context,
+   typename field_indices<AH>::type const& indexIn,
+   typename field_indices<bar<dq>::type>::type const& indexOut1,
+   typename field_indices<dq>::type const& indexOut2
+   ) const
+{
+   // get HBBbar vertex
+   // we don't use amplitude_squared here because we need both this vertex
+   // both with running and pole masses
+   const auto indices = concatenate(indexOut1, indexOut2, indexIn);
+   const auto HBBbarVertexDR = Vertex<bar<dq>::type, dq, AH>::evaluate(indices, context);
+
+   const double mAOS = context.physical_mass<AH>(indexIn);
+   const double flux = 1./(2.*mAOS);
+
+   constexpr double color_factor = squared_color_generator<AH, bar<dq>::type, dq>();
+
+   if(!boost::range::equal(indexOut1, indexOut2)) {
+      if (!is_zero(HBBbarVertexDR.left()) || !is_zero(HBBbarVertexDR.right())) {
+         const double mdqOS1 = context.physical_mass<dq>(indexOut1);
+         const double mdqOS2 = context.physical_mass<dq>(indexOut2);
+         const auto xOS1 = Sqr(mdqOS1/mAOS);
+         const auto xOS2 = Sqr(mdqOS2/mAOS);
+         const double phase_space = 1./(8.*Pi) * std::sqrt(KallenLambda(1., xOS1, xOS2));
+         return flux * phase_space * color_factor * amplitude_squared<AH, bar<dq>::type, dq>(context, indexIn, indexOut1, indexOut2);
+      }
+      return 0.;
+   }
+
+   const double mdqDR = context.mass<dq>(indexOut1);
+   const double mdqOS = context.physical_mass<dq>(indexOut1);
+   if(is_zero(mdqDR) || is_zero(mdqOS)) {
+      throw std::runtime_error("Error in A->ddbar: down quarks cannot be massless");
+   }
+   const auto xOS = Sqr(mdqOS/mAOS);
+   const auto xDR = Sqr(mdqDR/mAOS);
+
+   // TODO: add off-shell decays?
+   if (4.*std::max(xDR, xOS) > 1.) {
+      return 0.;
+   }
+
+   const auto betaOS = std::sqrt(1.-4.*xOS);
+   const auto betaDR = std::sqrt(1.-4.*xDR);
+
+   const double phase_spaceDR = 1./(8.*Pi) * std::sqrt(KallenLambda(1., xDR, xDR));
+   const double phase_spaceOS = 1./(8.*Pi) * std::sqrt(KallenLambda(1., xOS, xOS));
+
+   const auto HBBbarVertexDR_P = 0.5*(HBBbarVertexDR.right() - HBBbarVertexDR.left());
+
+   double amp2DR_P = 0;
+   double amp2OS_P = 0;
+      amp2DR_P = Sqr(mAOS) *
```

Comment:
Please fix indentation here. :)

Reply:
Done in 7abe348909868d91930ceae808bf3071b4e153f1